### ISSUE TITLE
Documentation for writing custom validators

### ISSUE THREAD

**Issue by filleduchaos** [stablekernel/aqueduct#285]
Hi,

I'm not sure if this is the right place to raise this issue, but there's a problem with the [docs for custom validators](http://aqueduct.io/docs/db/validations/#other-validator-behavior)

When overriding the `validate` method in the `ManagedObject` class the first call should be to `super.validate()`, not `super()`

So the snippet
```
class Person extends ManagedObject<_Person> implements _Person {
  @override
  bool validate(
      {ValidateOperation forOperation: ValidateOperation.insert,
      List<String> collectErrorsIn}) {
   var valid = super(
     forOperation: forOperation, collectErrorsIn: collectErrorsIn);

    if (a + b > 10) {
      valid = false;
      collectErrorsIn.add("a + b must be greater than 10");
    }

    return valid;
  }
}
```
should be
```
class Person extends ManagedObject<_Person> implements _Person {
  @override
  bool validate(
      {ValidateOperation forOperation: ValidateOperation.insert,
      List<String> collectErrorsIn}) {
   var valid = super.validate(
     forOperation: forOperation, collectErrorsIn: collectErrorsIn);

    if (a + b > 10) {
      valid = false;
      collectErrorsIn.add("a + b must be greater than 10");
    }

    return valid;
  }
}
```

**Comment by joeconwaystk** [comment]
:+1: Thanks, will make this change soon. If you are up to contributing, you can also issue PRs against `docs/source` branch. The user guides are in the `source/docs` directory.

**Comment by filleduchaos** [comment]
@joeconwaystk You're welcome! I did raise a PR.

**Comment by joeconwaystk** [comment]
Awesome, thank you much! And thanks for the callout on the API reference on dart docs, too. I'll get these merged and they'll go out with the 2.3.0 release.

**Comment by joeconwaystk** [comment]
Now available on the main site, thanks
